runtime/v2/manager_test.go

```diff
@@ -0,0 +1,108 @@
+//go:build !windows && !darwin
+// +build !windows,!darwin
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package v2
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+// setupAbsoluteShimPath creates a temporary directory with an empty shim executable file in it.
+// This is to test the exec.LookPath branch of resolveRuntimePath
+func setupAbsoluteShimPath() (string, error) {
+	tempShimDir, err := ioutil.TempDir("", "test-resolve-runtime-path")
+	if err != nil {
+		return "", err
+	}
+
+	_, err = os.Create(tempShimDir + "/containerd-shim-runc-v2")
+	if err != nil {
+		return "", err
+	}
+
+	err = os.Setenv("PATH", tempShimDir+":"+os.Getenv("PATH"))
```

Comment:
It is bit scary to set the PATH here since it will affect all tests in the same process. How about using t.Setenv?